core-authentication/src/main/java/gg/solarmc/loader/authentication/AuthenticationCenter.java

```diff
@@ -69,59 +164,67 @@ public InitialLoginAttempt attemptInitialLogin(Transaction transaction, Unauthen
 				.fetchOne();
 		if (passwordRecord == null) {
 			if (user.isPremium()) {
-				// New premium user, insert empty password and permit login
+				// New premium username, insert empty password and permit login
 				insertAutomaticAccount(transaction, user);
-				return InitialLoginAttempt.premiumPermitted();
+				return LoginAttempt.premiumPermitted();
 			}
-			// New cracked user, request account creation
-			return InitialLoginAttempt.needsAccount();
+			// New cracked username, request account creation
+			return LoginAttempt.needsAccount();
 		}
+		LoggerFactory.getLogger(getClass()).info(
```

Comment:
This produces a new logger every time? Can't you just have a single defined logger? A bit nitpicky, i know, so feel free to disregard.